feat: Implement category list query API

zerone1202 opened the pull request:
## #️⃣ Issue Number
Closes #39

## 📝 요약(Summary)
- 카테고리 전체 목록 조회 기능을 구현
- 삭제되지 않은 카테고리를 name 기준 오름차순 정렬하여 반환
- 전체 사용자에게 공개되도록 Security 설정을 수정

## 🛠️ PR 유형
- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가 및 리팩토링
- [ ] 빌드 혹은 패키지 매니저 수정
- [ ] 파일 또는 폴더명 수정
- [ ] 파일 또는 폴더 삭제

## ✅ PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성

zerone1202 committed:
feat(category): Add CategoryGetAllResponse DTO

- 카테고리 전체 조회를 위한 응답 DTO를 추가
- categoryId, name 필드 포함

zerone1202 committed:
feat(category): Add method for category list retrieval in QueryService

- 카테고리 전체 조회 기능을 위한 메서드를 추가
- getAllCategories() 정의
- 반환 타입: List<CategoryGetAllResponse>

zerone1202 committed:
feat(category): Implement category list retrieval logic in QueryServiceImpl

- 카테고리 전체 조회 비즈니스 로직을 구현
- 삭제되지 않은 카테고리 조회
- name 오름차순 정렬 적용
- CategoryGetAllResponse로 변환하여 반환

zerone1202 committed:
feat(category): Add repository method for active category list retrieval

- 카테고리 전체 조회를 위한 Repository 메서드를 추가
- findAllByIsDeletedFalseOrderByNameAsc() 추가
- soft delete 필드 기준으로 필터링

zerone1202 committed:
feat(category): Add success message for category list fetching

- 카테고리 목록 조회 성공 메시지를 추가
- CATEGORY_LIST_FETCHED 상수 정의

zerone1202 committed:
feat(category): Add getAllCategories method to CategoryController interface

- 카테고리 전체 조회 엔드포인트 정의를 추가
- 반환 타입: ResponseEntity<CommonApiResponse<List<CategoryGetAllResponse>>>

zerone1202 committed:
feat(category): Implement category list retrieval API in Controller

- 카테고리 전체 조회 API를 구현
- GET /api/v1/categories
- CategoryQueryService 연동
- CommonApiResponse.success() 적용

zerone1202 committed:
feat(security): Allow public access to GET /api/v1/categories

- 카테고리 전체 조회 API 접근 권한을 public으로 허용
- GET /api/v1/categories → permitAll

zerone1202 committed:
Merge branch 'dev' of https://github.com/sparta-finale/golfrental into feat/39-category
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

zerone1202 commented on src/main/java/com/golfRental/domain/category/dto/response/CategoryGetAllResponse.java:
```diff
@@ -0,0 +1,10 @@
+package com.golfRental.domain.category.dto.response;
+
+import lombok.Builder;
+
+@Builder
+public record CategoryGetAllResponse(
+        Long categoryId,
+        String name
+) {
+}
```
저희 팀컨벤션은 record와 @Builder 사용으로 넘어가겠습니다.

zerone1202 commented on src/main/java/com/golfRental/domain/category/service/query/CategoryQueryServiceImpl.java:
```diff
@@ -0,0 +1,36 @@
+        List<Category> categories =
+                categoryRepository.findAllByDeletedAtIsNullOrderByNameAsc();
+
+        return categories.stream()
+                .map(category ->
+                        CategoryGetAllResponse.builder()
+                                .categoryId(category.getId())
+                                .name(category.getName())
+                                .build()
+                )
+                .toList();
```
추후 팀 논의 후 리팩토링시 반영하겠습니다.

zerone1202 commented on src/main/java/com/golfRental/domain/category/service/query/CategoryQueryService.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface CategoryQueryService {
+
+    // 카테고리 목록 조회
```
참고하겠습니다.

moominIsCute commented on src/main/java/com/golfRental/security/config/SecurityConfig.java:
```diff
@@ -51,6 +51,14 @@ public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Excepti
+                        // 관리자 전용 API
+                        .requestMatchers(HttpMethod.POST,
+                                "/api/v1/admin/**"
+                        ).hasRole("ADMIN")
```
선언 순서대로 실행됩니다. 구체적인 패턴이 먼저 오는것은 어떨까요?

https://velog.io/@jun0922/Spring-Security-Url-%ED%8C%A8%ED%84%B4-%EC%9A%B0%EC%84%A0%EC%88%9C%EC%9C%84

zerone1202 replied:
좋은 의견 감사합니다! 반영했습니다~

zerone1202 committed:
Merge branch 'dev' of https://github.com/sparta-finale/golfrental into feat/39-category
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

AllSungho reviewed (approved):
고생하셨습니다!!

moominIsCute commented:
> 고생하셨습니다!!

아직 리뷰중입니다.....;;

moominIsCute reviewed (commented):
이거 안보일까요?

zerone1202 committed:
chore(security): Reorder security URL matching priority

- Spring Security URL 패턴 매칭 우선순위를 올바르게 적용하기 위해 설정을 재정렬
- 회원가입/로그인 API를 최상단 permitAll로 배치
- 카테고리 전체 조회(GET /api/v1/categories) 공개 처리
- /open 엔드포인트 공개 처리
- 관리자 API(/api/v1/admin/**) 권한 규칙을 permitAll보다 상위에 배치
- /test 엔드포인트 ADMIN 권한 부여
- anyRequest().authenticated()를 가장 마지막에 배치하여 broad rule 충돌 방지

moominIsCute commented:
고생하셨습니다.
